No type definitions for Inflight classes `@wingcloud/framework`

### I tried this:

```typescript 
import { cloud, lift, main } from "@wingcloud/framework";
import * as slack from "@winglibs/slack";

main((root) => {
  let botToken = new cloud.Secret(root, "BotToken", {name: "bot-token"});
  let inbox = new cloud.Bucket(root, "Inbox");
  let slackBot = new slack.App(root, "SlackBot", { token: botToken});

  inbox.onCreate(lift({ slackBot }).inflight(async ({ slackBot }, key) => {
    console.log("Creating channel");
    let channel = await slackBot.channel("check-the-bot");
    channel.post(`New file: ${key} was uploaded`);
  }));
});
```

### This happened:

```sh
❯ wing compile main.ts
TS compilation failed:
main.ts:12:13 - error TS2339: Property 'post' does not exist on type 'Channel'.

12     channel.post(`New file: ${key} was uploaded`);
               ~~~~
```

### I expected this:

Compilation to be successful

### Is there a workaround?

I can cast Channel as any and it will work

```typescript
import { cloud, lift, main } from "@wingcloud/framework";
import * as slack from "@winglibs/slack";

main((root) => {
  let botToken = new cloud.Secret(root, "BotToken", {name: "bot-token"});
  let inbox = new cloud.Bucket(root, "Inbox");
  let slackBot = new slack.App(root, "SlackBot", { token: botToken});

  inbox.onCreate(lift({ slackBot }).inflight(async ({ slackBot }, key) => {
    console.log("Creating channel");
    let channel = await slackBot.channel("check-the-bot") as any; // THE CAST
    channel.post(`New file: ${key} was uploaded`);
  }));
});
```

### Anything else?

_No response_

### Wing Version

0.74.24

### Node.js Version

20

### Platform(s)

MacOS

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

--- a/libs/wingc/src/dtsify/mod.rs
+++ b/libs/wingc/src/dtsify/mod.rs
@@ -310,9 +310,7 @@ impl<'a> DTSifier<'a> {
 			}
 			StmtKind::Class(class) => {
 				code.line(self.dtsify_class(class, false));
-				if class.phase == Phase::Preflight {
-					code.line(self.dtsify_class(class, true));
-				}
+				code.line(self.dtsify_class(class, true));
 			}
 
 			// No need to emit anything for these

--- a/libs/wingc/src/dtsify/mod.rs
+++ b/libs/wingc/src/dtsify/mod.rs
@@ -202,7 +202,7 @@ impl<'a> DTSifier<'a> {
 			code.line(format!("constructor({constructor_params});"));
 		}
 
-		if !as_inflight && matches!(class.phase, Phase::Preflight) {
+		if !as_inflight {
 			// Emit special preflight marks to tie the inflight side together
 			let inflight_class_name = format!("{class_name}{TYPE_INFLIGHT_POSTFIX}");
 			code.line(format!(

--- a/libs/wingc/src/dtsify/mod.rs
+++ b/libs/wingc/src/dtsify/mod.rs
@@ -416,8 +416,18 @@ pub interface ClassInterface {
 	inflight bar(): void;
 }
 
+pub inflight interface InflightInterface {
+  somethingFun(): void;
+}
+
+pub inflight class InflightClass impl InflightInterface {
+  pub somethingFun() {}
+}
+
 pub class ParentClass impl ClassInterface {
-	pub static inflight static_method() {}
+	pub static inflight static_method(): InflightClass {
+	  return new InflightClass();
+	}
 
 	inflight foo() {}
 	pub inflight bar() {}
@@ -425,9 +435,8 @@ pub class ParentClass impl ClassInterface {
 	pub addHandler(handler: inflight (str): str) {}
 }
 
-pub class Child extends ParentClass impl ClassInterface {
-	
-}"#
+pub class Child extends ParentClass impl ClassInterface {}
+"#
 	);
 }
 

